lib/Sema/TypeCheckMacros.cpp

```diff
@@ -1049,8 +1049,10 @@ swift::expandFreestandingMacro(MacroExpansionDecl *med) {
 
   auto topLevelItems = macroSourceFile->getTopLevelItems();
   for (auto item : topLevelItems) {
-    if (auto *decl = item.dyn_cast<Decl *>())
+    if (auto *decl = item.dyn_cast<Decl *>()) {
+      decl->getAttrs().add(med->getAttrs());
       decl->setDeclContext(dc);
```

Comment:
This will cause problems when the same attribute is added to multiple places. Per our offline discussion, I think it's best for us to syntactically add the attributes (e.g., by splicing them into the `DeclSyntax` nodes we get back from the macro) rather than do it semantically. That way, we'll see the attributes in the expansion.

Reply:
I scoped down this PR to `Parse` support only. Removed this bit.